src/main/java/io/github/thebusybiscuit/exoticgarden/PlantsListener.java

```diff
@@ -50,12 +51,72 @@ public PlantsListener(ExoticGarden plugin) {
 
     @EventHandler
     public void onGrow(StructureGrowEvent e) {
+        if (PaperLib.isPaper()) {
+            if (PaperLib.isChunkGenerated(e.getLocation())) {
+                onGrowSf(e);
+            } else {
+                PaperLib.getChunkAtAsync(e.getLocation()).thenRun(() -> onGrowSf(e));
+            }
+        } else {
+            if (!e.getLocation().getChunk().isLoaded()) {
+                e.getLocation().getChunk().load();
+            }
+            onGrowSf(e);
+        }
+    }
+
+    @EventHandler
+    public void onGenerate(ChunkPopulateEvent e) {
+        if (!SlimefunPlugin.getWorldSettingsService().isWorldEnabled(e.getWorld())) {
+            return;
+        }
+
+        if (!cfg.getStringList("world-blacklist").contains(e.getWorld().getName())) {
+            Random random = ThreadLocalRandom.current();
+            if (random.nextInt(100) < cfg.getInt("chances.BUSH")) {
+                Berry berry = ExoticGarden.getBerries().get(random.nextInt(ExoticGarden.getBerries().size()));
+                if (berry.getType().equals(PlantType.ORE_PLANT)) return;
+
+                int x = (e.getChunk().getX() * 16 + random.nextInt(16));
+                int z = e.getChunk().getZ() * 16 + random.nextInt(16);
+
+                if ((x < 29999983 && x > -29999983) && (z < 29999983 && z > -29999983)) {
+                    if (PaperLib.isPaper()) {
+                        if (PaperLib.isChunkGenerated(e.getWorld(), x, z)) {
+                            blockSet(e, x, z, berry, random, true);
+                        } else {
+                            PaperLib.getChunkAtAsync(e.getWorld(), x, z).thenRun(() -> blockSet(e, x, z, berry, random, true));
+                        }
+                    } else {
+                        blockSet(e, x, z, berry, random, false);
+                    }
+                }
+            } else if (random.nextInt(100) < cfg.getInt("chances.TREE")) {
+                Tree tree = ExoticGarden.getTrees().get(random.nextInt(ExoticGarden.getTrees().size()));
+
+                int x = e.getChunk().getX() * 16 + random.nextInt(16);
+                int z = e.getChunk().getZ() * 16 + random.nextInt(16);
+
+                if ((x < 29999983 && x > -29999983) && (z < 29999983 && z > -29999983)) {
+                    if (PaperLib.isPaper()) {
+                        if (PaperLib.isChunkGenerated(e.getWorld(), x, z)) {
+                            pasteTree(e, x, z, tree);
+                        } else {
+                            PaperLib.getChunkAtAsync(e.getWorld(), x, z).thenRun(() -> pasteTree(e, x, z, tree));
+                        }
+                    } else {
+                        plugin.getServer().getScheduler().scheduleSyncDelayedTask(plugin, () -> pasteTree(e, x, z, tree));
+                    }
+                }
+            }
+        }
+    }
+
+    private void onGrowSf(StructureGrowEvent e) {
```

Comment:
This method name feels quite off to be honest, I'd suggest a more meaningful name such as `growStructure` since the `onGrow` name might make it easy to be confused with the EventHandler from above.